File: CPP/Komputer.cpp
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"


using namespace std;


class Komputer
{
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;
public:
    Komputer()
    {


    }


    Komputer(string nama, Cpu cpu, vector<Ram> ramList, Harddrive harddrive)
    {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddrive = harddrive;
    }


    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setCpu(Cpu cpu)
    {
        this->cpu = cpu;
    }


    void setRam(vector<Ram> ramList)
    {
        this->ramList = ramList;
    }    

    void setHarddrive(Harddrive harddrive)
    {
        this->harddrive = harddrive;
    }


    void addRam(Ram ram)
    {
        this->ramList.push_back(ram);
    }


    string getNama()
    {
        return this->nama;
    }


    Cpu getCpu()
    {
        return this->cpu;
    }


    vector<Ram> getRamList() // return vector
    {
        return this->ramList;
    }


    Harddrive getHarddrive()
    {
        return this->harddrive;
    }


    ~Komputer()
    {


    }
};

File: PY/Komputer.py
from Cpu import Cpu
from Ram import Ram
from Harddrive import Harddrive

class Komputer:
    __nama = ""
    __cpu = Cpu()
    __ram = []
    __harddrive = Harddrive()

    # inisialisasi sedikit berbeda, karena tidak bisa langsung inisialisasi atribut yang diberikan dari suatu class. 
    def __init__(self, nama="", cpu=Cpu(), ram=[], harddrive=Harddrive()):
        self.__nama = nama
        self.__cpu = cpu
        self.__ram = ram
        self.__harddrive = harddrive
    
    # Getter methods
    def getNama(self):
        return self.__nama
    
    def getCpu(self):
        return self.__cpu
    
    def getRam(self):
        return self.__ram
    
    def getHarddrive(self):
        return self.__harddrive
    
    # Setter methods
    def setNama(self, nama):
        self.__nama = nama
    
    def setCpu(self, cpu):
        self.__cpu = cpu
    
    def setRam(self, ram):
        self.__ram = ram
    
    def setHarddrive(self, harddrive):
        self.__harddrive = harddrive
    
    # Additional method to add a single RAM object to the list
    def addRam(self, ram):
        self.__ram.append(ram)

